### PR TITLE
Improve/addons hooks

### PR DESCRIPTION
**Pull request by ndelangen**
Migrate some addons to hooks

### CODE REVIEW

File: addons/a11y/src/components/A11YPanel.tsx [hunk 1]
```diff
@@ -90,187 +89,161 @@ type A11YPanelState =
...
-            });
-          }
-        }, 900);
+    setTimeout(() => {
+      const { status } = state;
+      if (status === 'ran') {
+        setState({
+          ...(state as RanState),
+          status: 'ready',
+        });
```
Comment (jonspalmer) [inline addons/a11y/src/components/A11YPanel.tsx]: why does this change from being called as the 2nd parameter of setState?

**Comment by jonspalmer** [comment]
I like it. I've overlapped with you here #9951 on the removing of some of the deprecated cruft.

**Comment by ndelangen** [comment]
@jonspalmer I'll close this PR in favour of yours, and you cam perhaps cherry-pick some of my changes?

**Comment by jonspalmer** [comment]
I didn't do any of the hooks upgrade. Maybe keep it open for that?

**Pull request closed without merging**